### PR TITLE
feat: file handling better testing

### PR DESCRIPTION
**Pull request by ben-kugler**
- Revert "Merge pull request #279 from momentohq/fix/revert-file-handling"
- Make sure we check for ini files before we try and read from them :)
- manually tested case where `cache/session-token` folder and file don't exist, we now get
   `ERROR: failed to get token for profile default, please run 'momento configure' to configure your profile`
in response

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,9 +6,9 @@ cargo build
 
 ### Testing
 
-Make sure you have `~/.momento/credentials` and `~/.momento/config` files with the following data.
+Make sure you have `~/.momento/cache/session-tokens` and `~/.momento/config` files with the following data.
```
Comment (cprice404) [inline CONTRIBUTING.md:9]: Is this actually required for testing?  This sounds really fragile and potentially destructive.  We need to fix whatever tests require this IMO.  Does not have to be in this PR.

File: momento/src/config.rs [hunk 2]
```diff
@@ -1,15 +1,62 @@
...
+
+    pub fn new_from_duration<S>(token: S, valid_for: std::time::Duration) -> Self
+    where
+        S: Into<String>,
+    {
+        let valid_for = match chrono::Duration::from_std(valid_for) {
+            Ok(inner) => inner,
+            Err(_) => panic!("Hit out of range error, this shouldn't have happend"),
+        };
+        let valid_for = (Utc::now() + valid_for).timestamp();
```
Comment (cprice404) [inline momento/src/config.rs:46]: Isn't this actually `valid_until`?

File: momento/src/utils/file.rs [hunk 3]
```diff
@@ -1,157 +1,276 @@
...
-pub async fn write_to_file(path: &str, file_contents: String) -> Result<(), CliError> {
-    let mut file = match fs::File::create(path).await {
-        Ok(f) => f,
-        Err(e) => {
-            return Err(CliError {
-                msg: format!("failed to write to file {path}, error: {e}"),
-            })
-        }
-    };
+fn get_credentials_file_path() -> Result<PathBuf, CliError> {
```
[the remark this answers is not in the document]
Comment (ben-kugler) [reply momento/src/utils/file.rs:104]: Oh, I didn't think that was the goal, but makes a lot of sense, I'll make those changes.

[the remark this answers is not in the document]
Comment (cprice404) [reply momento/src/utils/file.rs:104]: We should verify with @rlinehan , it's possible I'm out-of-date on the most recent design decisions.  But that was what I was expecting it to end up like.

[the remark this answers is not in the document]
Comment (rlinehan) [reply momento/src/utils/file.rs:104]: Yeah, what Chris described is i think also what I was expecting.

I think we should also make sure to test a few scenarios:
- a new CLI installation where user does `login` and it all works
- someone who already has credentials in their file, upgrades and everything continues to work as expected
- someone who already has credentials in their file, upgrades, everything continues to work as expected AND they try using `login` and it also works and then they can use the session token (potentially with profiles? I don't remember where we ended up on that, IIRC there was a design doc somewhere?)

[the remark this answers is not in the document]
Comment (ben-kugler) [reply momento/src/utils/file.rs]: So validated that
1. Current credentials will work as is
2. You can add new tokens with custom profiles in `credentials` file
3. `login` command creates a new token in `cache/session-tokens`

```
Order of credentials lookup
1. get credentials with user supplied profile
2. get credentials with user default profile
3. get credentials with momento login profile
```

File: momento/src/utils/file.rs [hunk 4]
```diff
@@ -65,17 +74,36 @@ async fn create_file(path: &PathBuf) -> Result<(), CliError> {
...
-    read_ini(&creds_path).await
+    let session_token_path = get_credentials_file_path()?;
+    read_ini_with_custom_default(
+        &session_token_path,
+        Some(LOGIN_DEFAULT_SESSION_PROFILE_NAME),
+    )
+    .await
+}
+
+pub async fn read_user_session_token_ini() -> Result<Ini, CliError> {
```
[the remark this answers is not in the document]
Comment (ben-kugler) [reply momento/src/utils/file.rs:89]: Oh geeze, yeah, that's just bad naming on my part. Will update to something like `read_user_account_token_ini`, assuming we are sticking with the `momento account` command.

**Pull request closed without merging**